### PR TITLE
[479137]improved Misleading error message 'Cannot create datatype' for terminal rule #91

### PR DESCRIPTION
**Pull request by cdietrich**
[479137]improved Misleading error message 'Cannot create datatype' for terminal rule #91

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=479137

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java [hunk 1]
```diff
@@ -1016,12 +1019,20 @@ private EClassifierInfo createEClassifierInfo(TypeRef typeRef, String name) thro
...
 				classifier = EcoreFactory.eINSTANCE.createEClass();
-			else if (GrammarUtil.containingEnumRule(typeRef) != null)
+			} else if (GrammarUtil.containingEnumRule(typeRef) != null) {
 				classifier = EcoreFactory.eINSTANCE.createEEnum();
-			else
-				throw new TransformationException(TransformationErrorCode.NoSuchTypeAvailable, "Cannot create datatype " + classifierName, typeRef);
+			} else {
+				for (AbstractMetamodelDeclaration mmd : grammar.getMetamodelDeclarations()) {
+					if (mmd instanceof ReferencedMetamodel) {
+						if (mmd.getEPackage() != null && mmd.getEPackage().getNsURI().equals(EcorePackage.eNS_URI)) {
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java]: condition could be appended to preceding if condition

Comment (cdietrich) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java]: upated

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java [hunk 2]
```diff
@@ -18,6 +18,8 @@
 import java.util.Map;
 import java.util.Set;
 
+import javax.print.attribute.standard.Media;
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java]: Whoot?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java [hunk 3]
```diff
@@ -1016,12 +1019,18 @@ private EClassifierInfo createEClassifierInfo(TypeRef typeRef, String name) thro
...
 				classifier = EcoreFactory.eINSTANCE.createEEnum();
-			else
-				throw new TransformationException(TransformationErrorCode.NoSuchTypeAvailable, "Cannot create datatype " + classifierName, typeRef);
+			} else {
+				for (AbstractMetamodelDeclaration mmd : grammar.getMetamodelDeclarations()) {
+					if (mmd instanceof ReferencedMetamodel && mmd.getEPackage() != null && mmd.getEPackage().getNsURI().equals(EcorePackage.eNS_URI)) {
+						throw new TransformationException(TransformationErrorCode.NoSuchTypeAvailable, "Cannot create datatype " + classifierName, typeRef);
+					}
+				}
+				throw new TransformationException(TransformationErrorCode.NoSuchTypeAvailable, "Cannot create datatype " + classifierName + ". Make sure you have imported '"+EcorePackage.eNS_URI+"'", typeRef);
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformer.java]: Maybe add 'If this is supposed to return EString ...'

**Pull request merged**